drivers/dma/dma_stm32_v1.c

```diff
@@ -35,7 +35,7 @@ uint32_t dma_stm32_id_to_stream(uint32_t id)
 	return stream_nr[id];
 }
 
-#if !defined(CONFIG_DMAMUX_STM32)
+#if !defined(CONFIG_SOC_SERIES_STM32H7X) && !defined(CONFIG_SOC_SERIES_STM32MP1X)
```

Comment:
Then, what about applying the same to `dma_stm32_slot_to_channel()` declaration in header:
https://github.com/zephyrproject-rtos/zephyr/blob/5fd0412d1db9018103d4814e8245948f2e0d4b43/drivers/dma/dma_stm32.h#L53-L55

[Out of scope of this PR]
This also raises the question on another user of this function: `stm32_dma_config_channel_function`, also defined under `#if !defined(CONFIG_DMAMUX_STM32)`, but this one is never called, so maybe it could be purely removed ?
@FRASTM can you have a look ?